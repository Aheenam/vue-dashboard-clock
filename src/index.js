import Clock from './components/Clock';

export default {
    install (Vue) {
        Vue.component('clock', Clock);
    }
}

export { Clock };
