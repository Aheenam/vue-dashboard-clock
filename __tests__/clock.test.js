
import { Clock } from '../src/index'
import Vue from 'vue'
import lolex from 'lolex'

describe('dashboard-test', () => {

  let Constructor = Vue.extend(Clock)
  let clock = null

  beforeEach(() => {
    clock = lolex.install({
      now: new Date('2018-01-01 12:00:00')
    })
  })

  it('should use timestamp if set', () => {

    let vm = createVm({
      timezone: 'AMERICA/LOS_ANGELES'
    })

    expect(vm.timezone).toEqual('AMERICA/LOS_ANGELES')
    expect(vm.dateTime.format('HH:mm:ss')).toEqual('03:00:00')

  })

  it('should run time', async () => {

    let vm = createVm({
      timezone: 'AMERICA/LOS_ANGELES'
    })

    clock.tick(1000)
    await Vue.nextTick()

    expect(vm.timezone).toEqual('AMERICA/LOS_ANGELES')
    expect(vm.dateTime.format('HH:mm:ss')).toEqual('03:00:01')

  })

  it('can render formatted date by props', () => {

    let vm = createVm({
      dateFormat: 'DD/MM/YYYY'
    })

    expect(vm.formattedDate).toEqual('01/01/2018')
    expect(vm.$el).toMatchSnapshot()

  })

  it('can render formatted time by props', () => {

    let vm = createVm({
      timeFormat: 'HH:mm'
    })

    expect(vm.formattedTime).toEqual('12:00')
    expect(vm.$el).toMatchSnapshot()

  })

  function createVm(propsData = {}) {

    return new Constructor({
      propsData: propsData
    }).$mount()

  }

})
