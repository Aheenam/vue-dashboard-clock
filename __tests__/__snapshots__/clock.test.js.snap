// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard-test can render formatted date by props 1`] = `
<div
  class="clock"
>
  <div
    class="date"
  >
    
    01/01/2018
  
  </div>
   
  <div
    class="time"
  >
    
    12:00:00
  
  </div>
   
  <div
    class="timezone"
  >
    
    Europe / Berlin
  
  </div>
</div>
`;

exports[`dashboard-test can render formatted time by props 1`] = `
<div
  class="clock"
>
  <div
    class="date"
  >
    
    Mon 01/01
  
  </div>
   
  <div
    class="time"
  >
    
    12:00
  
  </div>
   
  <div
    class="timezone"
  >
    
    Europe / Berlin
  
  </div>
</div>
`;
